src/Services/Azure/Authentication/CustomChainedCredential.cs

```diff
@@ -89,7 +90,16 @@ private static InteractiveBrowserCredential CreateBrowserCredential(string? tena
             brokerOptions.ClientId = clientId;
         }
 
-        return new(brokerOptions);
+        var browserCredential = new InteractiveBrowserCredential(brokerOptions);
+
+        // Check for timeout value in the environment variable
+        string? timeoutValue = Environment.GetEnvironmentVariable(BrowserAuhtenticationTimeoutEnvVarName);
```

Comment:
Does it make sense to always have a timeout with some default value, like 5 minutes? If someone uses your MCP and hits this, it would be hard to recover because they will need to search for the process and kill it.

Reply:
5 minute timeout as default, sounds fine to me.
we can adjust it, according to user feedback, as it comes.
the env var is still there to change it.